contrib/nydusify/pkg/generater/generater.go

```diff
@@ -0,0 +1,115 @@
+package generater
+
+import (
+	"context"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+
+	"github.com/dragonflyoss/image-service/contrib/nydusify/pkg/parser"
+	"github.com/dragonflyoss/image-service/contrib/nydusify/pkg/provider"
+	"github.com/dragonflyoss/image-service/contrib/nydusify/pkg/utils"
+)
+
+// Opt defines Chunkdict generate options.
+// Note: sources is one and more Nydus image references.
+type Opt struct {
+	WorkDir        string
+	Sources        string
+	SourceInsecure bool
+	NydusImagePath string
+	NydusdPath     string
+	ExpectedArch   string
+}
+
+// Chunkdict generate deposit multiple Nydus images into database
+// by Nydus-image to save image chunk and blob into database.
+type Generater struct {
+	Opt
+	sourceSlice  []string
+	sourceParser []*parser.Parser
+}
+
+// New creates Generater instance, sourceSlice is multiple Nydus images reference list.
```

Comment:
Please consider moving the comment for the variable `sourceSlice` to the line above it.